integration-libs/s4-service/root/facade/checkout-service-schedule-picker.service.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable, inject } from '@angular/core';
+import { BaseSiteService, CxDatePipe, TimeUtils } from '@spartacus/core';
+import { take } from 'rxjs';
+import { ServiceOrderConfig } from '../model/checkout-service-details.model';
+
+@Injectable()
+export class CheckoutServiceSchedulePickerService {
+  protected baseSiteService = inject(BaseSiteService);
+  protected datePipe = inject(CxDatePipe);
+
+  getMinDateForService(): string {
+    const config = this.getServiceOrderConfiguration();
+    let minDate = new Date();
+    minDate.setDate(minDate.getDate() + ((config?.leadDays ?? 0) + 1)); // min date is the date after the no.of lead days
+    return this.datePipe.transform(minDate, 'yyyy-MM-dd') ?? '';
+  }
+
+  getScheduledServiceTimes(): string[] {
+    return this.getServiceOrderConfiguration()?.serviceScheduleTimes ?? [];
+  }
+
+  getServiceOrderConfiguration(): ServiceOrderConfig | undefined {
```

Comment:
Here you subscribe to an observable but you also immediately return config. This could lead to a race condition where the method returns a value before the observable has emitted a value.

I believe this should work for you, please verify.
```
  getServiceOrderConfiguration(): Observable<ServiceOrderConfig | undefined> {
    return this.baseSiteService
      .get()
      .pipe(
        take(1),
        map((baseSite) => {
          let config = { leadDays: 0, serviceScheduleTimes: [] as string[] };
          config.leadDays =
            baseSite?.baseStore?.serviceOrderConfiguration?.leadDays ?? 0;
          config.serviceScheduleTimes =
            baseSite?.baseStore?.serviceOrderConfiguration
              ?.serviceScheduleTimes ?? ([] as string[]);
          return config;
        })
      );
  }
```